src/main/java/com/klarna/hiverunner/StandaloneHiveServerContext.java

```diff
@@ -50,228 +42,240 @@
-        conf.setBoolVar(METASTORE_VALIDATE_COLUMNS, true);
-        conf.setBoolVar(METASTORE_VALIDATE_TABLES, true);
+  protected void overrideHiveConf(HiveConf hiveConf) {
+    for (Map.Entry<String, String> hiveConfEntry : hiveRunnerConfig.getHiveConfSystemOverride().entrySet()) {
+      hiveConf.set(hiveConfEntry.getKey(), hiveConfEntry.getValue());
     }
-
-    private void configureDerbyLog() {
-          // overriding default derby log path to not go to root of project
-          File derbyLogFile;
-          try {
-              derbyLogFile = File.createTempFile("derby", ".log");
-              LOGGER.debug("Derby set to log to " + derbyLogFile.getAbsolutePath());
-          } catch (IOException e) {
-              throw new UncheckedIOException("Error creating temporary derby log file", e);
-          }
-          System.setProperty("derby.stream.error.file", derbyLogFile.getAbsolutePath());
-    }
-
-    protected void configureFileSystem(Path basedir, HiveConf conf) throws IOException {
-        conf.setVar(METASTORECONNECTURLKEY, metaStorageUrl + ";create=true");
-
-        createAndSetFolderProperty(METASTOREWAREHOUSE, "warehouse", conf, basedir);
-        createAndSetFolderProperty(LOCALSCRATCHDIR, "localscratchdir", conf, basedir);
-        createAndSetFolderProperty(HIVEHISTORYFILELOC, "tmp", conf, basedir);
-
-        conf.setBoolVar(HIVE_WAREHOUSE_SUBDIR_INHERIT_PERMS, true);
-
-        createAndSetFolderProperty("hadoop.tmp.dir", "hadooptmp", conf, basedir);
-        createAndSetFolderProperty("test.log.dir", "logs", conf, basedir);
-
-
-
-        /*
-            Tez specific configurations below
-         */
-        /*
-            Tez will upload a hive-exec.jar to this location.
-            It looks like it will do this only once per test suite so it makes sense to keep this in a central location
-            rather than in the tmp dir of each test.
-         */
-        File installation_dir = newFolder(basedir, "tez_installation_dir").toFile();
-
-        conf.setVar(HiveConf.ConfVars.HIVE_JAR_DIRECTORY, installation_dir.getAbsolutePath());
-        conf.setVar(HiveConf.ConfVars.HIVE_USER_INSTALL_DIR, installation_dir.getAbsolutePath());
-    }
-
-    Path newFolder(Path basedir, String folder) throws IOException {
-        Path newFolder = Files.createTempDirectory(basedir, folder);
-        FileUtil.setPermission(newFolder.toFile(), FsPermission.getDirDefault());
-        return newFolder;
-    }
-
-    @Override
-    public HiveConf getHiveConf() {
-        return hiveConf;
-    }
-
-    @Override
-    public Path getBaseDir() {
-        return basedir;
-    }
-
-    protected final void createAndSetFolderProperty(HiveConf.ConfVars var, String folder, HiveConf conf, Path basedir)
-        throws IOException {
-        conf.setVar(var, newFolder(basedir, folder).toAbsolutePath().toString());
+  }
+
+  protected void configureMrExecutionEngine(HiveConf conf) {
```

Comment:
this might not be necessary anymore since Hive3 dropped M/R support. But not a blocker happy to merge this in for now.

Reply:
Yeah, good point, I think we leave it for now, we could have a follow on PR that goes through the code and removes everything to do with MR as there's probably more than just this.